examples/policies/multi-asset-policy/multi-asset-policy.pact

```diff
@@ -76,29 +83,33 @@
   (defun propose-asset:bool (token-id:string uri:string)
     (enforce-non-fungible-policy token-id)
     (enforce (not (= uri "")) "URI cannot be empty")
-    (enforce-guard (at 'guard (read token-operators token-id)))
     (let* (
       (assets:[string] (+ (get-proposed-assets token-id) [uri]))
-      (asset-id:integer (- (length assets) 1)) 
+      (asset-id:integer (- (length assets) 1))
+      (operator-guard:guard (at 'guard (read token-operators token-id)))
     )
-      (write proposed-assets token-id { 'assets: assets })
-      (emit-event (ASSET_PROPOSED token-id asset-id uri))
-      true
+      (with-capability (ASSET_PROPOSED token-id asset-id uri operator-guard)
```

Comment:
For this I recommend looking up `operator-guard` inside `ASSET_PROPOSED` capability, because UI wise, the signer has to look up and provide operator-guard in the capability list